### ISSUE TITLE
Gold Tracking in Progress Bar

### ISSUE THREAD

**Issue by WanderingFox** [tullamods/Dominos#495]
**Is your feature request related to a problem? Please describe.**
It's hard to visualize gold progress towards a goal like it is experience, reputation, honor etc.

While it's not a main concern for leveling characters, working towards a goal amount of currency is the primary reason some people play wow (AH goblins, bank alts, etc)

**Describe the solution you'd like**
It'd be wonderful to be able to visualize gold gains in the same way that experience is shown (as a progress bar). Ideally this would present in the same 'stacked bars' system as xp+rested xp. In other words, 'max' on the bar would be some settable goal amount (say 1million gold) the 'value' of the bar would be the amount of gold on the current character as well as (optionally) the total goal on the server (eg. from something like altoholic, broker_currency, tsm, collected via dominos, etc).

In other words:
```
[   player gold   |   total gold on the server   |                                       ] goal amount
```

**Describe alternatives you've considered**
Other tracking systems all provide a numerical amount, but don't visualize it. Dominos is in a unique position to show it as if it is a progress bar (which it is)

**Additional context**
It was pretty easy to stick together a concept of current player gold / goal amount
![image](https://user-images.githubusercontent.com/2967926/97833825-499a5f00-1ca4-11eb-8b26-65bd7335f0a0.png)

**Comment by addonaddict** [comment]
I would love it if this idea was implemented!

**Comment by WanderingFox** [comment]
Had some more time to work on this and have a functioning prototype. Cleaning it up so that it actually has options and isn't hard-coded everything, but once that's done I'll submit a PR to hopefully get it included :)

It uses Altoholic's DataStore addon to calculate gold on realm if it's available and shows it as 'rested xp'

![image](https://user-images.githubusercontent.com/2967926/98481310-61ec0b80-21c7-11eb-908c-c83cc61f4c8b.png)

**Comment by WanderingFox** [comment]
@addonaddict Found some time to implement this. Hopefully @Tuller has a moment sometime soon to consider it :)
